lib/vuln/php_easter_egg/engine.py

```diff
@@ -0,0 +1,201 @@
+                            '?=PHPE9568F35-D428-11d2-A769-00AA001ACF42',
+                            '?=PHPE9568F36-D428-11d2-A769-00AA001ACF42',]
+    }
+
+
+def conn(targ, port, timeout_sec, socks_proxy):
+    try:
+        if socks_proxy is not None:
+            socks_version = socks.SOCKS5 if socks_proxy.startswith(
+                'socks5://') else socks.SOCKS4
+            socks_proxy = socks_proxy.rsplit('://')[1]
+            if '@' in socks_proxy:
+                socks_username = socks_proxy.rsplit(':')[0]
+                socks_password = socks_proxy.rsplit(':')[1].rsplit('@')[0]
+                socks.set_default_proxy(socks_version, str(socks_proxy.rsplit('@')[1].rsplit(':')[0]),
+                                        int(socks_proxy.rsplit(':')[-1]), username=socks_username,
+                                        password=socks_password)
+                socket.socket = socks.socksocket
+                socket.getaddrinfo = getaddrinfo
+            else:
+                socks.set_default_proxy(socks_version, str(socks_proxy.rsplit(':')[0]),
+                                        int(socks_proxy.rsplit(':')[1]))
+                socket.socket = socks.socksocket
+                socket.getaddrinfo = getaddrinfo()
+        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sys.stdout.flush()
+        s.settimeout(timeout_sec)
+        s.connect((targ, port))
+        return s
+    except Exception:
+        return None
+
+
+def php_easter_egg_vuln(target, port, timeout_sec, log_in_file, language, time_sleep,
+                   thread_tmp_filename, extra_requirements, socks_proxy, scan_id, scan_cmd):
+    try:
+        try:
+            s = conn(target, port, timeout_sec, socks_proxy)
+        except Exception:
+            return False
+        if not s:
+            return False
+        else:
+            user_agent_list = useragents.useragents()
+            global php_easter_egg
+            if target_type(target) != "HTTP" and port == 443:
+                target = 'https://' + target
+            if target_type(target) != "HTTP" and port == 80:
+                target = 'http://' + target
+            if(not target.endswith("/")):
+                req_php_credits_url = target + "/" + extra_requirements["php_easter_eggs"][0]
+                req_php_logo_url = target + "/" + extra_requirements["php_easter_eggs"][1]
+                req_zend_url = target + "/" + extra_requirements["php_easter_eggs"][2]
+                req_php_logo2_url = target + "/" + extra_requirements["php_easter_eggs"][3]
+            try:
+                user_agent = {'User-agent': random.choice(user_agent_list)}
+                req_php_credits = requests.get(req_php_credits_url, timeout=10, headers=user_agent)
+                user_agent = {'User-agent': random.choice(user_agent_list)}
+                req_php_logo = requests.get(req_php_logo_url, timeout=10, headers=user_agent)
```

Comment:
@aman566 please do not hardcode the timeout - it should be timeout_sec also verify=False is missing